src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -3205,47 +3194,31 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
+        [OuterLoop("Uses external server")]
         [Fact]
-        public async Task Proxy_UseSecureProxyTunnel_Success()
+        public async Task Proxy_SendSecureRequestThruProxy_ConnectTunnelUsed()
         {
-            if (IsWinHttpHandler || IsNetfxHandler || IsCurlHandler)
-            {
-                // Issue #27746: WinHttpHandler and netfx hang on this test. 
-                // The same happens consistently on macOS 10.13 Release and with some
-                // frequency on some Linux flavors, disabling the test for curl handler due to that.
-                return;
-            }
-
-            LoopbackServer.Options options = new LoopbackServer.Options { UseSsl = true };
-
             var listener = new TcpListener(IPAddress.Loopback, 0);
             listener.Start();
             var ep = (IPEndPoint)listener.Server.LocalEndPoint;
             Uri proxyUrl = new Uri($"http://{ep.Address}:{ep.Port}/");
-            //TODO : refactor once LoopbackGetRequestHttpProxy is merged with LoppbackServer
-            Task proxy = LoopbackGetRequestHttpProxy.StartAsync(listener, false, false);
+            _output.WriteLine($"proxy uri = {proxyUrl.ToString()}");
+            // TODO : refactor once LoopbackGetRequestHttpProxy is merged with LoopbackServer
+            Task<LoopbackGetRequestHttpProxy.ProxyResult> proxyTask =
+                LoopbackGetRequestHttpProxy.StartAsync(listener, false, false);
 
-            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            HttpClientHandler handler = CreateHttpClientHandler();
+            handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl);
+            using (var client = new HttpClient(handler))
+            using (HttpResponseMessage response = await client.GetAsync(Configuration.Http.SecureRemoteEchoServer))
             {
-                Task serverTask = server.AcceptConnectionAsync(async connection =>
-                {
-                    await connection.ReadRequestHeaderAndSendResponseAsync(content: "OK\r\n");
-                });
-
-                using (HttpClientHandler handler = CreateHttpClientHandler()){
-                    // Point handler at out loopback proxy
-                    handler.Proxy = new UseSpecifiedUriWebProxy(proxyUrl, null);
-                    handler.ServerCertificateCustomValidationCallback = TestHelper.AllowAllCertificates;
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            }
 
-                    using (HttpClient client = new HttpClient(handler))
-                    {
-                        HttpResponseMessage response = await client.GetAsync(url);
-                        Assert.True(response.StatusCode ==  HttpStatusCode.OK);
-                    }
-               }
-               await serverTask;
-            }, options);
-            await proxy;
+            await TaskTimeoutExtensions.WhenAllOrAnyFailed(new Task[] { proxyTask }, TestHelper.PassingTestTimeoutMilliseconds);
```

Comment:
This isn't waiting on multiple tasks anymore, so the WhenAllOrAnyFailed isn't needed.  It can instead be:
```C#
await proxyTask.TimeoutAfter(TestHelper.PassingTestTimeoutMilliseconds);
```

Reply:
Thx.  I'll change that.